tests/integration/nm/wired_test.py

```diff
@@ -40,6 +45,25 @@ def test_interface_mtu_change(eth1_up):
     }
 
 
+def test_interface_mac_change_with_reapply_fails(eth1_up):
+    with pytest.raises(TestMainloopError):
+        _test_interface_mac_change()
```

Comment:
What is the benefit of this extra internal function? It seems to be specific to this test and even shares most of the name.

Reply:
It is a bit more readable but mainly because of the modify PR that follows.